lib/vector-core/src/event/metric.rs

```diff
@@ -981,7 +1059,36 @@ impl Display for Metric {
             MetricKind::Incremental => '+',
         };
         write!(fmt, "{} {} ", &self.series, kind)?;
-        match &self.data.value {
+        write!(fmt, "{}", &self.data.value)
```

Comment:
I believe you are indeed correct, so I've gone ahead and swapped to that, and also have done the same for `self.series` since it has a dedicated `Display` implementation now.